src/puppetlabs/trapperkeeper/services/status/status_core.clj

```diff
@@ -63,22 +82,28 @@
 
 (defn build-routes
   [status-fns-atom]
-  (compojure/routes
-    (compojure/context "/v1" []
-      (compojure/GET "/services" []
-        (let [statuses (call-status-fns status-fns-atom)]
-          {:status 200
-           :body statuses}))
-       (compojure/GET "/services/:service-name" [service-name]
-         (if-let [service-info (get (deref status-fns-atom) service-name)]
-           (let [status (call-latest-status-fn-for-service service-info)]
-             {:status 200
-              :body (assoc status :service-name service-name)})
-           {:status 404
-            :body {:error {:type :service-not-found
-                           :message (str "No status information found for service "
-                                         service-name)}}})))))
+  (handler/api
+    (compojure/routes
+      (compojure/context "/v1" []
+        (compojure/GET "/services" [:as {params :params}]
+          (let [level (get-status-detail-level params)
+                statuses (call-status-fns (deref status-fns-atom) level)]
```

Comment:
This is a little bit of a nitpick, and this wasn't introduced in this PR... but...

Looking over this namespace, it seems to me like the fact that there is an atom involved in our management of the state of the service is an implementation detail that doesn't necessarily need to bleed through here?

I *think* it should always be the case that the registration of the status callback functions for various services will happen during lifecycle init phases, and thus the atom will be effectively immutable by the time we want to build the routes... in which case it might be better to just pass a plain-old map in to the route building fns?  And to the `update` function at the top of this namespace?  Then the use of the atom would be restricted to the service code, and since it's kind of tightly coupled with the TK service context, that makes some sense to me and seems like it might be a bit cleaner.

If you decide that's worth doing, feel free to merge after that change goes in, you don't need to wait another day for me to look at it again.

If you decide it's not worth doing, I'm good with this being merged w/o that change.